### PR TITLE
Actually use MinGW on CI

### PR DESCRIPTION
**Pull request by Luthaf**
The MinGW job was also using MSVC compiler by mistake

# Contributor (creator of pull-request) checklist

 - [ ] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - [ ] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
 📚 [Download documentation preview for this pull-request](https://nightly.link/lab-cosmo/metatensor/actions/artifacts/1815307315.zip)

⚙️ [Download Python wheels for this pull-request (you can install these with pip)](https://nightly.link/lab-cosmo/metatensor/actions/artifacts/1686235583.zip)

### CODE REVIEW

**Comment by Luthaf** [comment]
Not working yet, the output still contains

```
-- Building for: Visual Studio 16 2019
-- Running with CMake version 3.29.3
-- The C compiler identification is MSVC 19.29.30154.0
-- The CXX compiler identification is MSVC 19.29.30154.0
```

File: metatensor-core/tests/cmake-project/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,12 @@ project(metatensor-test-cmake-project C CXX)
 
 option(USE_CMAKE_SUBDIRECTORY OFF)
 
+if (MINGW)
+    # CI can't find libsdc++, so we statically link it
```
Comment (PicoCentauri) [inline metatensor-core/tests/cmake-project/CMakeLists.txt:10]: Not only CI but always when trying to compile locally on 🪟 .

```suggestion
    # can't find libsdc++, so we statically link it
```

File: metatensor-core/tests/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,11 @@ if (${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
     endif()
 endif()
 
+if (MINGW)
+    # CI can't find libsdc++, so we statically link it
```
Comment (PicoCentauri) [inline metatensor-core/tests/cpp/CMakeLists.txt:16]:
```suggestion
    # can't find libsdc++, so we statically link it
```

**Review by PicoCentauri — COMMENTED** [review]
Develop on Windows. It will be fun, they say...

**Pull request merged**